### PR TITLE
Completed spring06

### PR DESCRIPTION
**Pull request by Vedroid**

### CODE REVIEW

**Review by Mandybur800 — APPROVED** [review]

File: src/main/java/ua/vedroid/cinema/security/AuthenticationService.java [hunk 1]
```diff
@@ -6,5 +6,5 @@
 public interface AuthenticationService {
     User login(String email, String password) throws AuthenticationException;
 
-    User register(String email, String password);
+    User register(User user);
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/security/AuthenticationService.java]: there is no need to change this
at the point of registration we don't have a concept of user yet - just login and password

File: src/main/java/ua/vedroid/cinema/controller/ShoppingCartController.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final MovieSessionService movieSessionService;
+    private final UserService userService;
+    private final ShoppingCartMapper mapper;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/controller/ShoppingCartController.java:24]: let's either use @Autowired annotation above the constructor everywhere - or not use it it all

File: src/main/java/ua/vedroid/cinema/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    public UserController(UserService service, UserMapper mapper) {
+        this.service = service;
+        this.mapper = mapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getByEmail(@RequestParam String email) {
+        return service.findByEmail(email).map(mapper::toDto).get();
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/controller/UserController.java:26]: nice 👍

File: src/main/java/ua/vedroid/cinema/model/dto/UserRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package ua.vedroid.cinema.model.dto;
+
+public class UserRequestDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/model/dto/UserRequestDto.java]: do we need this?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/ua/vedroid/cinema/security/AuthenticationService.java (hunk 1 above)
Comment (Vedroid) [reply src/main/java/ua/vedroid/cinema/security/AuthenticationService.java]: Если мы при регистрации будем указывать имя, номер телефона... ?

Comment (KseniiaMakarova) [reply src/main/java/ua/vedroid/cinema/security/AuthenticationService.java]: in this case yes, we may think about refactoring

File: src/main/java/ua/vedroid/cinema/controller/AuthenticationController.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import ua.vedroid.cinema.model.dto.UserRequestDto;
+import ua.vedroid.cinema.security.AuthenticationService;
+import ua.vedroid.cinema.service.mapper.UserMapper;
+
+@RestController
+public class AuthenticationController {
+    private final AuthenticationService service;
+    private final UserMapper mapper;
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/controller/AuthenticationController.java]: please remove this field

File: src/main/java/ua/vedroid/cinema/service/mapper/UserMapper.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+package ua.vedroid.cinema.service.mapper;
+
+import ua.vedroid.cinema.model.User;
+import ua.vedroid.cinema.model.dto.UserRequestDto;
+import ua.vedroid.cinema.model.dto.UserResponseDto;
+
+public interface UserMapper extends GenericDtoMapper<User, UserResponseDto>,
+        GenericEntityMapper<User, UserRequestDto> {
```
Comment (KseniiaMakarova) [inline src/main/java/ua/vedroid/cinema/service/mapper/UserMapper.java]:
```suggestion
public interface UserMapper extends GenericDtoMapper<User, UserResponseDto> {
```

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**